test/framework/src/main/java/org/elasticsearch/test/rest/section/ResponseBodyAssertion.java

```diff
@@ -0,0 +1,38 @@
+package org.elasticsearch.test.rest.section;
+
+import java.io.IOException;
+import java.util.Map;
+
+import org.elasticsearch.common.logging.ESLoggerFactory;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.json.JsonXContent;
+import org.elasticsearch.test.rest.parser.RestTestFragmentParser;
+import org.elasticsearch.test.rest.parser.RestTestParseException;
+import org.elasticsearch.test.rest.parser.RestTestSuiteParseContext;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Checks that the response body matches some text. 
+ */
+public class ResponseBodyAssertion extends Assertion {
+    public static final RestTestFragmentParser<ResponseBodyAssertion> PARSER = new RestTestFragmentParser<ResponseBodyAssertion>() {
+        @Override
+        public ResponseBodyAssertion parse(RestTestSuiteParseContext parseContext) throws IOException, RestTestParseException {
+            try (XContentParser parser = JsonXContent.jsonXContent.createParser(parseContext.parseField())) {
+                return new ResponseBodyAssertion("$body", parser.map());
+            }
+        }
+    };
+
+    private ResponseBodyAssertion(String field, Map<String, Object> expectedValue) {
+        super(field, expectedValue);
+    }
+
+    @Override
+    protected void doAssert(Object actualValue, Object expectedValue) {
+        // TODO Try and make it more clear what didn't match - sort or something.
+        assertThat("body didn't match the expected value", actualValue, equalTo(expectedValue));
```

Comment:
As is this could be replaced by an invocation to `match: $body` in most cases. I'm still torn on if we need this. It'd be nice to have nicer error messaging, but maybe I should just build that into match. I probably should.